### DISCUSSION TITLE
nunit-allure implementation

### DISCUSSION THREAD

**Discussion by pattaswamyr** [allure-framework/allure2#1824] [Questions & Support] [answered]
Hi Team,

We are implementing Nunit-Allure reports on existing C# framework.
In currently framework we are using Nunit reporting format.

I have gone through the documents to implement Nunit-Allure in existing project, but here some issue with framework structure.
We have multiple Nunit Test-Projects in framework as below.

Project-Solution 
         Engine
         Pages
         Test-Module-1
         Test-Module-2
         Test-Module-3
         ..
         Test-Module-N

All our test cases are distributed in the multiple Test-modules (projects) under one solution file.
Do I need to add Allure-Report folder in all Test-Projects ?
How to get a consolidated report after executing multiple test-cases from different modules.
Can someone help me, how to implement the same.

**Comment by neparij** [comment, marked as answer]
By design - every module has its own build directory with an allure-results folder inside.
But anyway, you can save your results to any directory you have write permissions.
Saving results to a **single** folder this way and generating report looks like a solution.
To change the allure-results directory, use allureConfig.json (it must be copied to the output directory on build)
```
{
  "allure": {
    "directory": "<YOUR-RESULTS-DIRECTORY-HERE>",
  }
}
```
